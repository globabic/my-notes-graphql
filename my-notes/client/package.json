{
  "name": "client",
  "version": "0.1.0",
  "babel": {
    "presets": [
      [
        "env",
        {
          "modules": false
        }
      ],
      "react"
    ],
    "env": {
      "test": {
        "presets": [
          [
            "env"
          ],
          "react"
        ]
      }
    }
  },
  "private": true,
  "dependencies": {
    "apollo-boost": "^0.1.10",
    "graphql": "^0.13.2",
    "react": "^16.4.1",
    "react-apollo": "^2.1.9",
    "react-dom": "^16.4.1",
    "react-scripts": "1.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "eject": "react-scripts eject"
  },
  "devDependencies": {
    "apollo-client-mock": "0.0.8",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.4",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^23.4.2",
    "react-test-renderer": "^16.4.1",
    "regenerator-runtime": "^0.12.0"
  },
  "jest": {
    "verbose": true,
    "testURL": "http://localhost:3000/",
    "transform": {
      "^.+\\.(jsx|js)?$": "<rootDir>/node_modules/babel-jest"
    },
    "moduleNameMapper": {
      "\\.css$": "identity-obj-proxy"
    },
    "collectCoverageFrom": [
      "**/*.component.{js,jsx}",
      "!**/node_modules/**"
    ],
    "coverageReporters": [
      "text-summary"
    ]
  }
}
