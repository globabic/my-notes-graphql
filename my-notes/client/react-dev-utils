+ ent@2.2.0
added 1 package in 16.24s
