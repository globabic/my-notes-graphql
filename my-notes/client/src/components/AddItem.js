import React, { Component } from 'react';
import {graphql, compose} from 'react-apollo';
import {addNoteMutation, getNotesQuery} from '../queries/queries';

import '../css/addItem.css';

class AddIteam extends Component {
  constructor(props){
     super(props)
      this.state = {
        content: '',
        date: ''
      };
   }//constructor
   submitForm(e){
     e.preventDefault();
     let value = this.contentInput.value;
     if(value !== null && value !== '' && value.length !== 0) {
       let date = new Date().toLocaleString('en-GB', {hour12: false}).split(',');
       this.props.addNoteMutation({
         variables:{
           content: value,
           date: date.toLocaleString()
         },
         refetchQueries: [{query: getNotesQuery}]
       });
       this.form.reset();
       this.contentInput.className = "";
       this.contentInput.placeholder = "Enter your note...";
       this.setState({
         content: '',
         date: ''
       });
     } else {
       this.contentInput.placeholder = "Content can't be empty";
       this.contentInput.className = "error";
     }

 }//submitForm

  render(){
    return(
      <form id="add-item" ref = {f => this.form = f} onSubmit={this.submitForm.bind(this)}>
          <input id="content" ref = {c => this.contentInput = c} placeholder="Enter your note..." type="text"/>
          <input type="submit" value="Add" />
      </form>
    )
  }//render
}

export default compose(
  graphql(getNotesQuery, {name: "getNotesQuery"}),
  graphql(addNoteMutation, {name: "addNoteMutation"})
)(AddIteam);
