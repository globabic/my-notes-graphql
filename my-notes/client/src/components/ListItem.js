import React, { Component } from 'react';
import {graphql, compose} from 'react-apollo';
import '../css/listItem.css';

import {deleteNoteMutation, editNoteMutation, getNotesQuery} from '../queries/queries';

class ListItem extends Component {
  constructor(props){
     super(props);
     this.state = {
      selected: null
    }
   }//constructor
  render(){
    return(
      <li>
        <div className="list-item">
            <img className="list-item-avatar" alt="avatar" src={require("../images/avatar.svg")}/>
            <div className="list-item-bubble">
                <span className="list-item-name">{this.props.item}</span>
                <img className="list-item-edit" src={require("../images/edit.svg")} alt="edit" onClick={this.handleEdit.bind(this)}></img>
                <img className="list-item-delete" src={require("../images/delete.svg")} alt="delete" onClick={this.handleDelete.bind(this)}></img>
            </div>
            <span className="list-item-date">{this.props.date}</span>
          </div>
      </li>
    )
  }//render
  handleDelete(){

    this.props.deleteNoteMutation({
      variables:{
        id: this.props.id
      },
      refetchQueries: [{query: getNotesQuery}]
    }
   );
 }//handleDelete
  handleEdit(){
    
    let newValue = prompt("Edit your note: ", this.props.item);
    let today = new Date();
    let date = today.getDate() + '/' + (today.getMonth() + 1) + '/' + today.getFullYear() + ' ' + today.getHours() + ':' + today.getMinutes();
    if(newValue != null){
      this.props.editNoteMutation({
        variables:{
          id: this.props.id,
          content: newValue,
          date: date
        },
        refetchQueries: [{query: getNotesQuery}]
      }
    );

    }

  }//handleEdit

}//ListItem

export default compose(
  graphql(getNotesQuery, {name: "getNotesQuery"}),
  graphql(deleteNoteMutation, {name: "deleteNoteMutation"}),
  graphql(editNoteMutation, {name: "editNoteMutation"})
)(ListItem);
