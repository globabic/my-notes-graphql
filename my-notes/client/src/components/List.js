import React, { Component } from 'react';
import {graphql} from 'react-apollo';
import {getNotesQuery} from '../queries/queries';

import '../css/list.css';

//modules
import ListItem from './ListItem';
import AddItem from './AddItem';



class List extends Component {
  displayNotes(){
    let data = this.props.data;
    if(data.loading){
      return(<div>Loading...</div>)
    }else if(data.notes === undefined || data.notes.length === 0){
      return(<div>Add a new note</div>)
    }else{
      return data.notes.map(note =>{
        return(
          <ListItem item={note.content} key={note.id} id={note.id} date={note.date} />
        )
      })
    }
  }//displayNotes
  countItems(){
      let data = this.props.data;
    if(data.loading){
      return "Calculating..."
    }else{
      let notes = data.notes;
      return notes ? notes.length : 0;
    }
  }
  render() {
  return (

      <div id="notes-list">
        <p>{this.countItems()} items</p>
        <ul>
          {this.displayNotes()}
        </ul>
        <AddItem />
      </div>


  );
}//render

}

export default graphql(getNotesQuery)(List);
