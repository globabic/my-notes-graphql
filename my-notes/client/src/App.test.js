import React from 'react';
import {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {render, mount} from 'enzyme'
import List from './components/List';
import ListItem from './components/ListItem';
import AddItem from './components/AddItem';
import ApolloClient from 'apollo-boost';
import {ApolloProvider} from 'react-apollo';
import client from './clientmock'

Enzyme.configure({ adapter: new Adapter() });

function pageBody(func) {
  return func(<ApolloProvider client={client()}>
  <div id="notes-list">
    <List>
    <div id="notes-list">
      <p>1 items</p>
      <ul>
        <ListItem item="Hello content" key="123" id="123" date="24/07/2018, 21:53:18" />
      </ul>
      <AddItem />
    </div>
    </List>
  </div>
  </ApolloProvider>)
}

describe('List', () => {
    test('renders no notes', () => {
        const resolverOverwrites = {
          Query: () => ({
            // empty
          })
        }
        const renderedComponent = render(
          <ApolloProvider client={client()}>
            <div id="notes-list">
              <List />
            </div>
          </ApolloProvider>
        );

        expect(renderedComponent).toMatchSnapshot();
    });

    test('renders one item', () => {
        const renderedComponent = pageBody(render);
        expect(renderedComponent).toMatchSnapshot();
    });

    test('error on empty submit', () => {
        const mounted = pageBody(mount)
        // fetch AddItem inner form
        const form = mounted.find("#add-item")
        form.simulate('submit', {
          preventDefault: () => {}
        })
        // check input placeholder in AddItem inner form
        expect(form.find("#content").render().prop("placeholder")).toEqual("Content can't be empty")
    });

    test('successful submit', () => {
        const mounted = pageBody(mount)
        // fetch AddItem inner form
        const form = mounted.find("#add-item")
        // set input value in form
        const input = form.find("#content")
        input.instance().value = 'My test note';

        form.simulate('submit', {
          preventDefault: () => {}
        })

        expect(input.render().prop("placeholder")).toEqual("Enter your note...")
    })
});
