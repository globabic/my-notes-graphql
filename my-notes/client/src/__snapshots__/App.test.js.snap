// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders no notes 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "id": "notes-list",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "id": "notes-list",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Calculating... items",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Loading...",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "ul",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "id": "add-item",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "id": "content",
                      "placeholder": "Enter your note...",
                      "type": "text",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "type": "submit",
                        "value": "Add",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "type": "submit",
                      "value": "Add",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "id": "content",
                        "placeholder": "Enter your note...",
                        "type": "text",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "id": undefined,
                        "placeholder": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "id": undefined,
                        "placeholder": undefined,
                        "type": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                ],
                "name": "form",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Loading...",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "ul",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "id": "add-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "id": "content",
                    "placeholder": "Enter your note...",
                    "type": "text",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "type": "submit",
                      "value": "Add",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "type": "submit",
                    "value": "Add",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "id": "content",
                      "placeholder": "Enter your note...",
                      "type": "text",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                },
              ],
              "name": "form",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "Calculating... items",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {
              "id": "add-item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "id": "content",
                  "placeholder": "Enter your note...",
                  "type": "text",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "type": "submit",
                    "value": "Add",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "type": "submit",
                  "value": "Add",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "id": "content",
                    "placeholder": "Enter your note...",
                    "type": "text",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "form",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Loading...",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "ul",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Calculating... items",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "id": undefined,
        },
        "x-attribsPrefix": Object {
          "id": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`List renders one item 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "id": "notes-list",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "id": "notes-list",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Calculating... items",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Loading...",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "ul",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "id": "add-item",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "id": "content",
                      "placeholder": "Enter your note...",
                      "type": "text",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "type": "submit",
                        "value": "Add",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "type": "submit",
                      "value": "Add",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "id": "content",
                        "placeholder": "Enter your note...",
                        "type": "text",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "id": undefined,
                        "placeholder": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "id": undefined,
                        "placeholder": undefined,
                        "type": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                ],
                "name": "form",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Loading...",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "ul",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "id": "add-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "id": "content",
                    "placeholder": "Enter your note...",
                    "type": "text",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "type": "submit",
                      "value": "Add",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "type": "submit",
                    "value": "Add",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "id": "content",
                      "placeholder": "Enter your note...",
                      "type": "text",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "id": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                },
              ],
              "name": "form",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "Calculating... items",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {
              "id": "add-item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "id": "content",
                  "placeholder": "Enter your note...",
                  "type": "text",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "type": "submit",
                    "value": "Add",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "type": "submit",
                  "value": "Add",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "id": "content",
                    "placeholder": "Enter your note...",
                    "type": "text",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "form",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Loading...",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
              ],
              "name": "ul",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Calculating... items",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "id": undefined,
        },
        "x-attribsPrefix": Object {
          "id": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
